locked-asset/proxy_dex/src/proxy_pair.rs

```diff
@@ -64,25 +61,26 @@ pub trait ProxyPairModule:
             lp_token_id: add_liq_result.lp_tokens_received.token_identifier.clone(),
             lp_token_amount: add_liq_result.lp_tokens_received.amount.clone(),
         };
+
+        let wrapped_lp_mapper = self.wrapped_lp_token();
         let token_merge_requested = !payments.is_empty();
-        let new_merged_attributes = if token_merge_requested {
-            let wrapped_lp_mapper = self.wrapped_lp_token();
+        let new_wrapped_token = if token_merge_requested {
             let wrapped_lp_tokens =
                 WrappedLpToken::new_from_payments(&payments, &wrapped_lp_mapper);
 
+            self.burn_multi_esdt(&payments);
+
             self.merge_wrapped_lp_tokens_with_virtual_pos(wrapped_lp_tokens, new_token_attributes)
-                .attributes
         } else {
-            new_token_attributes
-        };
+            let new_token_amount = new_token_attributes.get_total_supply().clone();
```

Comment:
don't we need to burn in this case as well ?

Reply:
Nope, the initial payments are Locked tokens, and another token. We need to keep the locked tokens to send them back on Remove Liquidity, and the other payment, is used for Add Liquidity.